### PR TITLE
Update RuleDetails for compact mode

### PR DESCRIPTION
**Pull request by LI0131**
Fixes some of the styling for compact mode

Before:

Screen Shot 2020-08-11 at 12 55 08 PM

Screen Shot 2020-08-11 at 12 55 29 PM

After:

Screen Shot 2020-08-11 at 12 54 03 PM
Screen Shot 2020-08-11 at 12 54 29 PM

### CODE REVIEW

File: src/PresentationalComponents/RuleDetails/_RuleDetails.scss [hunk 1]
```diff
@@ -63,3 +71,15 @@ hr {
     margin-bottom: var(--pf-global--spacer--md);
   }
 }
+
+@media only screen and (max-width: 768px) {
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: does this exist as a pf var? feels like we shouldn't be hardcoding this value 🤔

Comment (AllenBW) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: yep
`--pf-global--breakpoint--md | global_breakpoint_md | 768px`

File: src/SmartComponents/Recs/Details.scss [hunk 2]
```diff
@@ -17,3 +19,9 @@
 .categoryLabel {
   margin-left: var(--pf-global--spacer--md)
 }
+
+@media only screen and (max-width: 768px) {
```
Comment (AllenBW) [inline src/SmartComponents/Recs/Details.scss:23]: following comment above, set this this to `var(--pf-global--breakpoint--md)`

File: src/PresentationalComponents/RuleDetails/_RuleDetails.scss (hunk 1 above)
Comment (LI0131) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: So did a little digging on this... and it looks like it isn't possible use `var()` in a media query. https://stackoverflow.com/questions/40722882/css-native-variables-not-working-in-media-queries (unless we want to use a preprocessor?)

Comment (AllenBW) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: oooooo  what about `@media only screen and (min-width: $ins-break--md) `

Comment (AllenBW) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: actually you dont neet to do that `$ins-break--md: var(--pf-global--breakpoint--md);` is already defined in
@import '~@redhat-cloud-services/frontend-components-utilities/files/Utilities/_all';

Comment (LI0131) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: Oooo I had the same thought 😂 

I think that approach falls victim to the same issue though. The styles don't seem to apply 🤷‍♂️🤔

Comment (AllenBW) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: local definition as well?  🤔 I see the value hardcoded (not the pf var) in a few places.. would hate for us to also do it 😢 will take a look tomorrow 👍

Comment (AllenBW) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: great minds 🧠 😆

Comment (AllenBW) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss:75]: well... s on the radar https://github.com/w3c/csswg-drafts/issues/2627 but yeah not currently possible in the way   would hope

**Pull request merged**